src/go/rdctl/pkg/snapshot/snapshotter_unix.go

```diff
@@ -123,16 +123,25 @@ func (snapshotter SnapshotterImpl) CreateFiles(snapshot Snapshot) error {
 // to their working location.
 func (snapshotter SnapshotterImpl) RestoreFiles(snapshot Snapshot) error {
 	files := getSnapshotFiles(snapshotter.Paths, snapshot.ID)
+	var err error
 	for _, file := range files {
 		filename := filepath.Base(file.WorkingPath)
-		err := copyFile(file.WorkingPath, file.SnapshotPath, file.CopyOnWrite, file.FileMode)
+		err = copyFile(file.WorkingPath, file.SnapshotPath, file.CopyOnWrite, file.FileMode)
 		if errors.Is(err, os.ErrNotExist) && file.MissingOk {
-			if err := os.RemoveAll(file.WorkingPath); err != nil {
-				return fmt.Errorf("failed to remove %s: %w", filename, err)
+			if err = os.RemoveAll(file.WorkingPath); err != nil {
+				err = fmt.Errorf("failed to remove %s: %w", filename, err)
+				break
 			}
 		} else if err != nil {
-			return fmt.Errorf("failed to restore %s: %w", filename, err)
+			err = fmt.Errorf("failed to restore %s: %w", filename, err)
+			break
 		}
 	}
-	return nil
+	if err != nil {
```

Comment:
When the user runs `rdctl snapshot restore` from the command line, and an error occurs, should they get some kind of feedback that a factory reset has effectively been done? We now have this in the UI, but it's missing for the command line. I'm not sure this is the place to do it though. A `fmt.Printf` here may suffice?

Reply:
I don't think printing a message deep in the lower level function is a good idea. I think it would also interfere with the JSON output.

I thought about just wrapping `err` with another err that includes the info about the reset, but that would then also show up in the UI.

I think adding the message is outside the scope of this PR; we should create a new Github issue and deal with it when we refactor the code. I would like to have some kind of "flag" on the error to indicate that a data reset was performed, so it can be reported back via the JSON stream as well without having to parse the text of the error message.